### PR TITLE
test/extended: check if prometheus is firing a watchdog alert

### PR DESCRIPTION
**Pull request by paulfantom**
Checking if Watchdog alert is in firing state by querying prometheus metrics.

Based on `"should report telemetry if a cloud.openshift.com token is present"` test and on https://github.com/openshift/cluster-monitoring-operator/pull/266 PR.

/cc @brancz @mxinden @s-urbaniak @squat @metalmatze

### CODE REVIEW

File: test/extended/prometheus/prometheus.go [hunk 1]
```diff
@@ -150,6 +150,22 @@ var _ = g.Describe("[Feature:Prometheus][Conformance] Prometheus", func() {
...
 			})).NotTo(o.HaveOccurred())
 		})
+		g.It("should have a Watchdog alert in firing state", func() {
+			oc.SetupProject()
+			ns := oc.Namespace()
+			execPodName := e2e.CreateExecPodOrFail(oc.AdminKubeClient(), ns, "execpod", func(pod *v1.Pod) { pod.Spec.Containers[0].Image = "centos:7" })
+			defer func() { oc.AdminKubeClient().CoreV1().Pods(ns).Delete(execPodName, metav1.NewDeleteOptions(1)) }()
+
+			tests := map[string][]metricTest{
+				// should have constantly firing a watchdog alert
```
[the remark this answers is not in the document]
Comment (paulfantom) [reply test/extended/prometheus/prometheus.go]: I don't want to restart whole CI pipeline to fix this here, but I will fix it in #22513 as that one will need a rebase.

**Comment by mxinden** [comment]
Let us know if you need a _lgtm_ from our side. Not sure what the process is in this repository.

**Pull request merged**